test/VM.js

```diff
@@ -130,6 +137,24 @@ describe("VM", function () {
     console.log(`String concatenation: ${receipt.gasUsed.toNumber()} gas`);
   });
 
+  it("Should execute payable function", async () => {
+    const amount = appendDecimals(123);
+    const planner = new weiroll.Planner();
+
+    planner.add(payable.pay().withValue(amount));
+    const balance = planner.add(payable.balance());
+    planner.add(events.logUintPayable(balance));
+    const { commands, state } = planner.plan();
+
+    const tx = await vm.execute(commands, state, {value: amount});
+    await expect(tx)
+      .to.emit(eventsContract.attach(vm.address), "LogUint")
+      .withArgs(amount);
+
```

Comment:
To test this fully worked check the balance as well.

Reply:
Added one, it should be better